pinot-common/src/main/java/com/linkedin/pinot/common/utils/HLCSegmentNameHolder.java

```diff
@@ -45,11 +45,24 @@
   private final String _sequenceNumber;
   private final String _partitionRange;
   private final String _segmentName;
+  private final RealtimeSegmentType _segmentType;
+  private final String _tableName;
+
+  public static boolean isRealtimeV1Name(String segmentId) {
```

Comment:
No, the logic of associating the number of parts with a V1 style name belongs in the HLC class. If you notice, the only caller for this method is the parent class, which uses it to determine the segment name type.